### PR TITLE
Complete `pydantic/functional_validators.py` docstrings

### PR DESCRIPTION
**Pull request by hramezani**
Selected Reviewer: @samuelcolvin

### CODE REVIEW

**Comment by hramezani** [comment]
please review

File: pydantic/functional_validators.py [hunk 1]
```diff
@@ -388,16 +412,24 @@ def __call__(
...
+    def __call__(  # noqa: D102
         self: _ModelType,  # type: ignore
     ) -> _ModelType:
         ...
 
 
 class ModelAfterValidator(Protocol):
+    """A @model_validator decorated function signature.
+    This is used when `mode='after'`.
+    """
```
Comment (Kludex) [inline pydantic/functional_validators.py:429]: We have 120 line length 👀 
```suggestion
    """A @model_validator decorated function signature. This is used when `mode='after'`."""
```

File: pydantic/functional_validators.py [hunk 2]
```diff
@@ -388,16 +412,24 @@ def __call__(
 
 
 class ModelAfterValidatorWithoutInfo(Protocol):
+    """A @model_validator decorated function signature.
+    This is used when `mode='after'` and the function does not have info argument.
+    """
```
Comment (Kludex) [inline pydantic/functional_validators.py:417]:
```suggestion
    """A @model_validator decorated function signature. This is used when `mode='after'` and the function does not
    have info argument.
    """
```

File: pydantic/functional_validators.py [hunk 3]
```diff
@@ -375,7 +395,11 @@ def __call__(
 
 
 class ModelBeforeValidator(Protocol):
-    def __call__(
+    """A @model_validator decorated function signature.
+    This is used when `mode='before'`.
+    """
```
Comment (Kludex) [inline pydantic/functional_validators.py:400]:
```suggestion
    """A @model_validator decorated function signature. This is used when `mode='before'`."""
```

File: pydantic/functional_validators.py [hunk 4]
```diff
@@ -349,7 +361,11 @@ def __call__(
 
 
 class ModelWrapValidator(Protocol):
-    def __call__(
+    """A @model_validator decorated function signature.
+    This is used when `mode='wrap'`.
+    """
```
Comment (Kludex) [inline pydantic/functional_validators.py:366]:
```suggestion
    """A @model_validator decorated function signature. This is used when `mode='wrap'`."""
```

File: pydantic/functional_validators.py [hunk 5]
```diff
@@ -331,12 +333,22 @@ def dec(
 
 
 class ModelWrapValidatorHandler(_core_schema.ValidatorFunctionWrapHandler, Protocol[_ModelTypeCo]):
-    def __call__(self, input_value: Any, outer_location: str | int | None = None) -> _ModelTypeCo:  # pragma: no cover
+    """@model_validator decorated function handler argument type.
+    This is used when `mode='wrap'`.
+    """
```
Comment (Kludex) [inline pydantic/functional_validators.py:338]:
```suggestion
    """@model_validator decorated function handler argument type. This is used when `mode='wrap'`."""
```

**Review by Kludex — CHANGES_REQUESTED** [review]

**Review by Kludex — APPROVED** [review]

**Pull request merged**